src/lsdb/catalog/dataset/healpix_dataset.py

```diff
@@ -194,6 +194,22 @@ def _perform_search(
         ddf_partition_map = {pixel: i for i, pixel in enumerate(filtered_pixels)}
         return ddf_partition_map, filtered_partitions_ddf
 
+    def search(self, search: AbstractSearch):
```

Comment:
Can we delete the `search` method from the Catalog object entirely and just use this one to reduce duplication? We'd just need to be careful about margin filtering.

Reply:
We could also remove it from `MarginCatalog`:

https://github.com/astronomy-commons/lsdb/blob/34f7d9d518533f52e25f1d21d5db02eec26fefde/src/lsdb/catalog/margin_catalog.py#L28

Reply:
The catalog has the margin logic, but it should call the superclass, but yeah removing the margin is a good idea